Firmware/Chameleon-Mini/Application/MifareClassic.c

```diff
@@ -40,11 +40,17 @@
 
 #define ACK_NAK_FRAME_SIZE          4         /* Bits */
 #define ACK_VALUE                   0x0A
-/* Here are values from NXP MF1S50YYX_V1 datasheet, 9.3 */
+/* NAK values from NXP MF1S50YYX_V1 datasheet, 9.3 */
 #define NAK_S50_BUFFEROK_OPKO       0x00
 #define NAK_S50_BUFFEROK_PARITYCRC  0x01
-#define NAK_S50_BUFFERKO_OPKO       0x02
-#define NAK_S50_BUFFERKO_PARITYCRC  0X03
+#define NAK_S50_BUFFERKO_OPKO       0x04
+#define NAK_S50_BUFFERKO_PARITYCRC  0x05
```

Comment:
so you don't use 02 03 from the datasheet any longer?

Reply:
This was my typo :) 2 et 3 are not specifically defined in data sheets.
They were never used before (see original NAK defines), so that's even now.

My thought is that original old code is based on some observations on real life cards from KAOS, + some randomly chosen internal conventions then - there were clear comments in code that NAK codes to return had been chosen arbitrarily on some cases.
Everything which is not 0A is supposed to be taken as NAK anyway, according to older data sheets, so that would not really hurt.

Now for NAK numbers consistency, you can see NXP tried to do better in EV1 data sheet, by defining 4 cases, with 1 precise NAK number per case, still consistent with older data sheets though. This is still really blurry. We can see they tried to separate "CRC" errors from operation error (I suppose "invalid" operations), but "transfer buffer validity" condition is not very clear (if they are consistent with words, this would relate to TRANSFER/RESTORE/INC/DEC operations only, and state that the internal buffer is corrupted).

So I changed all NAK codes to be consistent with the most recent standard, but they remain consistent with old code (same NAK number for all CRC errors). The rest is just an internal condition to choose; I would take any, but think we should stick with only 2 codes: CRC invalid and operation invalid. We still have to choose if we want to consider "transfer buffer" is valid or no 🤣